parquet-hadoop/src/test/java/org/apache/parquet/hadoop/TestInterOpReadByteStreamSplit.java

```diff
@@ -67,7 +72,73 @@ public void testReadFloats() throws IOException {
             break;
         }
       }
-      assertTrue(reader.read() == null);
+      assertNull(reader.read());
     }
   }
+
+  private void compareColumnValues(
+      Path path, int expectRows, String leftCol, String rightCol, FieldComparator comparator) throws IOException {
+    try (ParquetReader<Group> reader =
+        ParquetReader.builder(new GroupReadSupport(), path).build()) {
+      for (int i = 0; i < expectRows; ++i) {
+        Group group = reader.read();
+        assertNotNull(group);
+        comparator.assertFieldsEqual(group, leftCol, rightCol);
+      }
+      assertNull(reader.read());
+    }
+  }
+
+  @Test
+  public void testReadAllSupportedTypes() throws IOException {
+    Path extendedFile = interop.GetInterOpFile(EXTENDED_FILE, CHANGESET);
+    final int expectRows = 200;
+    compareColumnValues(
+        extendedFile, expectRows, "float_plain", "float_byte_stream_split", ((group, leftCol, rightCol) -> {
+          float leftVal = group.getFloat(leftCol, 0);
+          float rightVal = group.getFloat(rightCol, 0);
+          assertEquals(leftVal, rightVal, 0.0);
+        }));
+    compareColumnValues(
```

Comment:
I'm curious if there's an easy way to reduce code repetition here. The various typed `get*` methods as well as the differing `assertEquals` signatures seem to get in the way.

Reply:
I'm curious too. Perhaps we can override GroupValueSource by adding `Object getObject(String field, int index)` which internally dispatches getXXX based on its GroupType?

Reply:
It appeats this triggers a check failure:
```
Error:  Failed to execute goal com.github.siom79.japicmp:japicmp-maven-plugin:0.18.2:cmp (default) on project parquet-column: There is at least one incompatibility: org.apache.parquet.example.data.Group:METHOD_ABSTRACT_ADDED_IN_SUPERCLASS,org.apache.parquet.example.data.GroupValueSource.getObject(int,int):METHOD_ABSTRACT_ADDED_TO_CLASS -> [Help 1]
```

Reply:
I guess the question is: is `org.apache.parquet.example.data` supposed to be a public API?

Reply:
Or I can simply remove the abstract method from the interface and cast directly to `SimpleGroup`.